core/src/main/scala/io/finch/ToResponse.scala

```diff
@@ -72,59 +79,60 @@ object ToResponse extends ToResponseInstances {
-      h: ToResponse.Aux[A, CTH],
-      t: Negotiable[A, CTT],
+    implicit def coproductToNegotiable[F[_], A, CTH <: String, CTT <: Coproduct](implicit
+      h: ToResponse.Aux[F, A, CTH],
+      t: Negotiable[F, A, CTT],
       a: Accept.Matcher[CTH]
-    ): Negotiable[A, CTH :+: CTT] = new Negotiable[A, CTH :+: CTT] {
-      def apply(accept: List[Accept]): ToResponse.Aux[A, CTH :+: CTT] =
-        if (accept.exists(_.matches[CTH])) h.asInstanceOf[ToResponse.Aux[A, CTH :+: CTT]]
-        else t(accept).asInstanceOf[ToResponse.Aux[A, CTH :+: CTT]]
+    ): Negotiable[F, A, CTH :+: CTT] = new Negotiable[F, A, CTH :+: CTT] {
+      def apply(accept: List[Accept]): ToResponse.Aux[F, A, CTH :+: CTT] =
+        if (accept.exists(_.matches[CTH])) h.asInstanceOf[ToResponse.Aux[F, A, CTH :+: CTT]]
+        else t(accept).asInstanceOf[ToResponse.Aux[F, A, CTH :+: CTT]]
     }
 
-    implicit def cnilToNegotiable[A, CTH <: String](implicit
-      tr: ToResponse.Aux[A, CTH]
-    ): Negotiable[A, CTH :+: CNil] = new Negotiable[A, CTH :+: CNil] {
-      def apply(accept: List[Accept]): ToResponse.Aux[A, CTH :+: CNil] =
-        tr.asInstanceOf[ToResponse.Aux[A, CTH :+: CNil]]
+    implicit def cnilToNegotiable[F[_], A, CTH <: String](implicit
+      tr: ToResponse.Aux[F, A, CTH]
+    ): Negotiable[F, A, CTH :+: CNil] = new Negotiable[F, A, CTH :+: CNil] {
+      def apply(accept: List[Accept]): ToResponse.Aux[F, A, CTH :+: CNil] =
+        tr.asInstanceOf[ToResponse.Aux[F, A, CTH :+: CNil]]
     }
 
-    implicit def singleToNegotiable[A, CT <: String](implicit
-      tr: ToResponse.Aux[A, CT]
-    ): Negotiable[A, CT] = new Negotiable[A, CT] {
-      def apply(accept: List[Accept]): ToResponse.Aux[A, CT] = tr
+    implicit def singleToNegotiable[F[_], A, CT <: String](implicit
+      tr: ToResponse.Aux[F, A, CT]
+    ): Negotiable[F, A, CT] = new Negotiable[F, A, CT] {
+      def apply(accept: List[Accept]): ToResponse.Aux[F, A, CT] = tr
     }
   }
 
-  trait FromCoproduct[C <: Coproduct] extends ToResponse[C]
+  trait FromCoproduct[F[_], C <: Coproduct] extends ToResponse[F, C]
 
   object FromCoproduct {
-    type Aux[C <: Coproduct, CT] = FromCoproduct[C] { type ContentType = CT }
+    type Aux[F[_], C <: Coproduct, CT] = FromCoproduct[F, C] { type ContentType = CT }
 
-    def instance[C <: Coproduct, CT <: String](fn: (C, Charset) => Response): Aux[C, CT] =
-      new FromCoproduct[C] {
+    def instance[F[_], C <: Coproduct, CT <: String](fn: (C, Charset) => F[Response]): Aux[F, C, CT] =
+      new FromCoproduct[F, C] {
         type ContentType = CT
-        def apply(c: C, cs: Charset): Response = fn(c, cs)
+        def apply(c: C, cs: Charset): F[Response] = fn(c, cs)
       }
 
-    implicit def cnilToResponse[CT <: String]: Aux[CNil, CT] =
-      instance((_, _) => Response(Version.Http10, Status.NotFound))
+    implicit def cnilToResponse[F[_], CT <: String](implicit
+      F: Effect[F]
```

Comment:
I think this could be relaxed to `Applicative`

Reply:
👍